device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -830,6 +776,10 @@ public void registerMultiplexedDeviceClient(List<DeviceClientConfig> configs, lo
                     }
 
                     registrationException.addRegistrationException(deviceId, deviceRegistrationException);
+
+                    // Since the registration failed, need to remove the device from the list of multiplexed devices
+                    DeviceClientConfig configThatFailedToRegister = this.deviceClientConfigs.remove(deviceId);
+                    ((AmqpsIotHubConnection) this.iotHubTransportConnection).unregisterMultiplexedDevice(configThatFailedToRegister);
```

Comment:
This is the actual code change that fixes a bug. This code sits within a block that waits while all the devices finish registration either successfully or unsuccessfully. The unsuccessful devices should be removed from this layer and from the AMQP layer so that their state matches that of the MultiplexingClient itself